src/backends/aws_s3.ts

```diff
@@ -34,9 +34,10 @@ const allowedMethods = ["GET", "HEAD"]
  * @param options AWS S3 bucket to proxy to
  * @module Backends
  */
-export function awsS3(options: AwsS3Options | string): FetchFunction {
+export function awsS3(options: AwsS3Options | string): ProxyFunction<AwsS3Options> {
```

Comment:
What's the difference here between a FetchFunction and a ProxyFunction?

Reply:
`ProxyFunction<T>` is a `FetchFunction` with the original config saved to a `proxyConfig` property on it. I'm not entirely sure why or if we're still using it but it's expected elsewhere and I didn't want to rip it out yet 😄